### ISSUE TITLE
GW permits in pydov

### ISSUE THREAD

**Issue by pjhaest** [DOV-Vlaanderen/pydov#284]
@Roel while working on the example for hydrogeology, I noticed that it would be really nice to have access to the GW permits with pydov as well. I know that it's just a wfs call as you already pointed out in #153  . But it's nicer if it would all fit in the same search.  
I tried to include it in pydov, but forgot that it is not on the same level as the other data sources. It does not have a Feature Catalogue, which seems a requirement for the data source to fit in the pydov AbstractSearch class. Before I start an arbitrary workaround with a separate definition of the object, do you think there would be convenient way to include this?

**Comment by pjhaest** [comment]
I will have a more thorough look this evening. Mabye add the _source=('wfs', 'xml')_ from the get_fields() abstract class method as attribute to the build_fields() abstract search method? This way, the 'xml'  can be left out when pointing to a WFS only layer?

**Comment by Roel** [comment]
Yes, a feature catalogue is required for a layer to work in pydov as this is where it gets the metadata of the available fields. I'm not in favour of dropping this requirement as it is good practice to document this for all layers anyway.

@jorissynaeve can you coordinate with VMM to provide a feature catalogue for the [Grondwatervergunningen (historische en huidige) layer](https://dov.vlaanderen.be/geonetwork/srv/eng/catalog.search#/metadata/1b8d9dce-40b2-450c-81ed-decb453b3143) and follow up here?

Once this is done I think we can add it in pydov, I believe it should be possible to create a type with only WfsFields?

**Comment by pjhaest** [comment]
Indeed, that is possible. The workaround would be to include a hardcoded feature catalogue object for that type in pydov. But I'm in favor of requesting a feature catalogue at the source.

**Comment by jorissynaeve** [comment]
> 
> 
> @jorissynaeve can you coordinate with VMM to provide a feature catalogue for the [Grondwatervergunningen (historische en huidige) layer](https://dov.vlaanderen.be/geonetwork/srv/eng/catalog.search#/metadata/1b8d9dce-40b2-450c-81ed-decb453b3143) and follow up here?


I've send an email to VMM in order to ask for creation of a FeatureCatalogue. As soon as I see progress, I will answer in this ticket.

**Comment by debaetpa** [comment]
Good news!
The feature catalogue for [Grondwatervergunningen (historische en huidige)](https://dov.vlaanderen.be/geonetwork/srv/eng/catalog.search#/metadata/1b8d9dce-40b2-450c-81ed-decb453b3143) is available

**Comment by pjhaest** [comment]
I already have part of the new class, I will finalize this as soon as possible

**Comment by Roel** [comment]
There appears to be only a feature catalog in production, not in oefen. Can the feature catalogue be added in oefen as well (tests are failing because of this)?

Another question: is there a permanent URL available for a grondwatervergunning? If so, why is it not included as a column in the WFS service? For consistency I'd like to have a permanent and unique URL for each object in pydov, and right now we only have an 'id' in the vergunningen WFS. There does exists a column 'installatie' with a permanent key of the installatie, but I'm assuming that's not unique among vergunningen?

**Comment by pjhaest** [comment]
OK, I understand that a URI (?) is the preferred way foward for pydov. @Roel would you not include this PR in pydov until that exists for grondwater vergunningen? Because I think it is something useful from a user-perspective.

**Comment by Roel** [comment]
If a permanent URI can be added in the nearish future, I'd like to wait for that. If not, we can work with the 'id' for now, but that requires extra code changes that would otherwise not be necessary.

I'm also wondering how this dataset relates to other objects we already have (grondwaterfilters f.ex.), is there a way to link a Grondwatervergunning to a Grondwaterfilter?

**Comment by debaetpa** [comment]
There is an permanent url for the permit, e.g. https://www.dov.vlaanderen.be/data/vergunning/2019-016520
But this url is not publicly available.
That is why the url was not added in the wfs.

We could introduce the permkey of the permit; however, that key does not uniquely identify a single record.
One gw permit consists of multiple installations (even for different 'rubrieken'), and for each installation there can be one or more 'deeltermijnen', each term associated with a different volume.

So the unique key for one record would be the combination of
- permkey_vergunning
- permkey_installatie
- vlaremrubriek
- van_datum_termijn
- tot_datum_termijn

Or you can use the single field 'id'.

**Comment by Roel** [comment]
Thanks for the explanation! i knew it was complicated but forgot the details..

I don't think it is a problem per se that the URL in not publicly available, the permkey_installatie is included in the WFS too and is not publicly available either.

We'll stick to the 'id' to uniquely identify a record, but wouldn't it be useful to add the permkey_vergunning to the WFS anyway? Can we know which records belong to the same permit without it (since it would be part of the unique key)?

**Comment by debaetpa** [comment]
No problem, we will add it to the wfs.

For now, you can use the 'watnr' to identify the vergunningen: same watnr = same permit

**Comment by debaetpa** [comment]
The feature catalog is now available in oefen.

**Comment by Roel** [comment]
One issue with the feature catalogue: apparently its uuid in oefen is different from production. We need it to be the same in oefen as it is in production, can this be fixed? Thx!

**Comment by debaetpa** [comment]
Ah, learnt something today.

Should be fixed now:
https://www.dov.vlaanderen.be/geonetwork/srv/dut/catalog.search#/metadata/5e605651-6b2e-406f-863f-d2eda4d3e534
https://oefen.dov.vlaanderen.be/geonetwork/srv/dut/catalog.search#/metadata/5e605651-6b2e-406f-863f-d2eda4d3e534

**Comment by debaetpa** [comment]
The field 'vergunning' was added tot the wfs.

**Comment by Roel** [comment]
Thanks! Tests are now passing.
